### PR TITLE
Rewrite the browser/connection/remotes-queue.js file to TypeScript

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
FPR

File: test/server/util-test.js [hunk 1]
```diff
@@ -72,10 +72,10 @@ describe('Utils', () => {
                 expected: {
                     name:            'Other',
                     version:         '0.0',
-                    platform:        'desktop',
-                    os:              { name: 'Windows', version: '0.0' },
+                    platform:        'other',
```
Comment (AndreyBelym) [inline test/server/util-test.js]: This test seems unrelated to the changes made by @miherlosev. Is it caused by a `bowser` update? 

BTW, do we really need tests for invalid/incomplete user agents? 😕 @Farfurix

Comment (miherlosev) [reply test/server/util-test.js]: Yes, it's change related to `bowser` module update.
Our tests check the default values for 'platform', 'version', etc.

Comment (Farfurix) [reply test/server/util-test.js]: @AndreyBelym @miherlosev I'll check this case

Comment (miherlosev) [reply test/server/util-test.js]: Now, I will merge this PR. Changes related to the tests will be done in the separate PR.
